newIDE/app/src/ProjectCreation/CreateProjectDialog.js

```diff
@@ -30,14 +30,14 @@ export type CreateProjectDialogProps = {|
 |};
 
 export type NewProjectSetup = {|
-  projectName: string,
   storageProvider: StorageProvider,
   saveAsLocation: ?SaveAsLocation,
-  height: number,
-  width: number,
-  orientation: 'landscape' | 'portrait' | 'default',
-  optimizeForPixelArt: boolean,
-  allowPlayersToLogIn: boolean,
+  projectName?: string,
+  height?: number,
+  width?: number,
+  orientation?: 'landscape' | 'portrait' | 'default',
+  optimizeForPixelArt?: boolean,
+  allowPlayersToLogIn?: boolean,
```

Comment:
Should you rather specify a type for new project setup with everything that is mandatory, and another type that corresponds to your needs?

Reply:
I thought about it, but at the same time, I feel that this type makes sense and can be reused if we need to create a project from somewhere else in the app. We can then specify the props we need